File: UseCases/src/test/resources/stub/OverloadProtocolTest.swift
var stubbedInt: Int! = 0
var int: Int {
set {
}
get {
return stubbedInt
}
}
var stubbedIntAddingResult: Int! = 0
func int(adding: Int) -> Int {
return stubbedIntAddingResult
}
func setValue(_ string: String, forKey key: String) {
}
func setValue(_ int: Int, forKey key: String) {
}
func set(value: String) {
}
func set(value: Int) {
}
var stubbedAnimateResult: Bool! = false
func animate() -> Bool {
return stubbedAnimateResult
}
func animate(withDuration duration: TimeInterval) {
}
func animate(withDuration duration: TimeInterval, delay: TimeInterval) {
}
func specialCharacters(_ tuple: (String, Int)) {
}
func specialCharacters(_ tuple: (UInt, Float)) {
}

File: UseCases/src/test/java/codes/seanhenry/mockgenerator/generator/templates/OverloadProtocolTest.kt
package codes.seanhenry.mockgenerator.generator.templates

import codes.seanhenry.mockgenerator.entities.Method
import codes.seanhenry.mockgenerator.entities.Property
import codes.seanhenry.mockgenerator.generator.MockTransformer

class OverloadProtocolTest : MockGeneratorTestTemplate() {

  override fun build(generator: MockTransformer) {
    generator.add(
        Property.Builder("int")
            .type("Int")
            .build()
    )
    generator.add(
        Method.Builder("int")
            .parameter("adding") { it.type("Int") }
            .returnType("Int")
            .build(),
        Method.Builder("setValue")
            .parameter("_", "string") { it.type("String") }
            .parameter("forKey", "key") { it.type("String") }
            .build(),
        Method.Builder("setValue")
            .parameter("_", "int") { it.type("Int") }
            .parameter("forKey", "key") { it.type("String") }
            .build(),
        Method.Builder("set")
            .parameter("value") { it.type("String") }
            .build(),
        Method.Builder("set")
            .parameter("value") { it.type("Int") }
            .build(),
        Method.Builder("animate")
            .returnType("Bool")
            .build(),
        Method.Builder("animate")
            .parameter("withDuration", "duration") { it.type("TimeInterval") }
            .build(),
        Method.Builder("animate")
            .parameter("withDuration", "duration") { it.type("TimeInterval") }
            .parameter("delay") { it.type("TimeInterval") }
            .build(),
        Method.Builder("specialCharacters")
            .parameter("_", "tuple") { it.type("(String, Int)") }
            .build(),
        Method.Builder("specialCharacters")
            .parameter("_", "tuple") { it.type("(UInt, Float)") }
            .build()
    )
  }
}
